Clean up manifest fallbacks section and add an extra caution

mattgarrish opened the pull request:
As mentioned in [#1464](https://github.com/w3c/epub-specs/issues/1464#issuecomment-1086242145), this PR separates the requirements for spine fallbacks from the requirements for content fallbacks.

I've also taken the technical details about the `fallback` attribute and merged them back into the `item` element definition, as this was the most bothersome part to me in moving this section up.

I've also added another layer of caution to the content fallbacks section that the mechanism was meant for `img` which now has other fallback mechanisms.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/epub-specs/pull/2203.html" title="Last updated on Apr 4, 2022, 4:56 PM UTC (2440b0f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/epub-specs/2203/5e0b77a...2440b0f.html" title="Last updated on Apr 4, 2022, 4:56 PM UTC (2440b0f)">Diff</a>

mattgarrish committed:
clean up manifest fallbacks section and add extra caution about content fallbacks

mattgarrish committed:
additional minor cleanup

mattgarrish committed:
move data url note to end of section

iherman commented on epub33/core/index.html:
```diff
@@ -1340,58 +1337,55 @@ <h5>Manifest Fallbacks</h5>
-
-						<p>EPUB Creators should use the intrinsic fallback capabilities of HTML and SVG to provide
-							fallback content whenever possible, and avoid using <a>Foreign Content Documents</a> in the
-							spine.</p>
```
I understand the reason for the caution but, spec-wise, it looks a bit strange to start a section with essentially saying "ignore this section". This may raise some question like "why do you have this section in the first place?" or "why is it not deprecated instead?". We know the answer but, nevertheless...

I think it looks somewhat less awkward if that caution is moved to the end of the section.

mattgarrish replied:
Or should we drop this one for now?

This is essentially pre-empting the "under-implemented" label which is going to tell people exactly the same thing. The caution about not needing to use manifest fallbacks for content anymore is a legitimate separate issue, but perhaps that only needs to be a note given it will also fall into the same general lack of support bucket.

iherman replied:
Yep, I think it is better to drop it for now. 

We will have to think about how we will, editorially, mark up under-implemented features... maybe something more than just a parenthetical word would be needed, We will see.

iherman commented on epub33/core/index.html:
```diff
@@ -1340,58 +1337,55 @@ <h5>Manifest Fallbacks</h5>
+										data-cite="html#the-img-element"><code>img</code> element</a>. As HTML now has
+									intrinsic fallback mechanism for images, the use of content fallbacks is strongly
+									discouraged. EPUB Creators should always use the intrinsic fallback capabilities of
+									[[HTML]] and [[SVG]] to provide fallback content.</p>
```
Same comment as above. I think it would be better to move this caution to the end of the `<dd>` part.

iherman commented on epub33/core/index.html:
```diff
@@ -1340,58 +1337,55 @@ <h5>Manifest Fallbacks</h5>
+						<dt id="content-fallbacks">Content Fallbacks</dt>
+						<dd>
+							<div class="caution">
+								<p>The purpose for keeping content fallbacks in EPUB 3 was primarily to continue to
```
```suggestion
								<p>The original reason for introducing fallbacks in EPUB 3 was primarily to continue to
```

The term "keeping" makes sense when one is familiar with the history (I presume it came from EPUB 2), but it sounds a bit strange for people who came into this world later.  Hence my proposal for change

mattgarrish replied:
But anyone who knows epub 2 knows that fallbacks didn't originate in epub 3, so it just reverses the confusion. 😄 

It also won't work logically to say we introduced fallbacks to continue to provide a means of specifying fallbacks for `img`. The precedent is missing. There was a small overlap between epub 2 and epub 3 when manifest fallbacks were the only way to handle `img`.

But to go off on a tangent, I was thinking of taking the two fallback sections currently under Foreign Resources and merging with manifest fallbacks under a "Resource Fallbacks" section. I'll ponder how to fix this note while I'm doing that.

mattgarrish replied:
I've modified this in the latest commit to:

> The original purpose for content fallbacks was to specify fallback images for 

Hopefully this works both ways. If you don't know the original purpose comes from EPUB 2, you can think the original purpose was for EPUB 3 and it changes nothing.

("Original purpose" knowingly forgets all about EPUB 2's use for "out-of-line xml islands", which were forgettable, and `object` which became `bindings` in EPUB 3 and then we gave up on in place of intrinsic content in the `object`.)

mattgarrish committed:
group intrinsic fallbacks into new "Resource Fallbacks" section;
remove caution note for manifest fallbacks and change caution on content fallbacks to a note and reword

mattgarrish committed:
Merge branch 'main' into restructure/fallbacks

mattgarrish committed:
reformat

mattgarrish committed:
Merge remote-tracking branch 'remotes/origin/main' into restructure/fallbacks
